src/main/main.js

```diff
@@ -54,18 +101,89 @@ export default class SbomGithubAction {
      * @returns {*} An object of secret and public arguments.
      */
     collectArguments() {
-        const secretParameters = ['sn-sbom-user', 'sn-sbom-password', 'sn-instance-url', 'gh-token'];
+        const secretParameters = [ 'sn-sbom-user', 'sn-sbom-password', 'sn-instance-url', 'gh-token' ];
         const secrets = secretParameters.reduce((acc, arg) => ({...acc, [arg]: core.getInput(arg)}), {});
-        const publicParameters = ['provider', 'repository', 'path', 'gh-account-owner' ];
+        const publicParameters = [ 'provider', 'repository', 'path', 'gh-account-owner' ];
         const parameters = publicParameters.reduce((acc, arg) => ({...acc, [arg]: core.getInput(arg)}), {});
-        return { ...secrets, ...parameters }
+        const uploadApiParameters = [
+            'business-application-id',
+            'business-application-name',
+            'product-model-id',
+            'requested-by',
+            'build-id',
+            'lifecycle-stage',
+            'fetch-vulnerability-info',
+            'fetch-package-info'
+        ];
+        const uploadApiArguments = uploadApiParameters.reduce((acc, arg) => ({...acc, [arg]: core.getInput(arg)}), {});
+        return { ...secrets, ...parameters, ...uploadApiArguments };
+    }
+
+    /**
+     * Performs the upload to the SBOM Workspace.
+     * @param document
+     * @param snInstanceUrl
+     * @param snUsername
+     * @param snPassword
+     * @returns {Promise<any>}
+     */
+    async upload({githubObjects, snInstanceUrl, snUsername, snPassword, uploadOptions}) {
+        let uploadUrl = this._generateSbomWorkspaceUrls({snInstanceUrl, apiPath: '/api/sbom/core/upload', uploadOptions});
+
+        let uploadPromises = githubObjects.map(({ document, file_name }) =>
+            fetch(uploadUrl, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Authorization': `Basic ${Buffer.from(snUsername + ":" + snPassword).toString('base64')}`
+                },
+                body: JSON.stringify(document)
+            })
+                .then(response => response.json())
+                .then(data => ({
+                    data, file_name
+                }))
+                .catch(error => {
+                    core.warning(`An error occurred while uploading SBOM: ${error.message}`)
+                    throw error;
+                })
+        )
+
+        return await Promise.all(uploadPromises).then(data => data.map(response => response));
+    }
+
+    _generateSbomWorkspaceUrls({snInstanceUrl, apiPath, uploadOptions}) {
+        let baseUrl = new URL(snInstanceUrl);
+        let requestUrl = new URL(apiPath, baseUrl);
+        Object.keys(uploadOptions).filter(key => uploadOptions[key]).forEach(key => requestUrl.searchParams.append(key, uploadOptions[key])); // TODO: double check filter condition
```

Comment:
is this checked?

Reply:
Yes, all inputs to GHA are strings. Therefore the filter will not interpret the string value of `"false"` as being equivalent to `false`. Consequently, the generated search parameters are `fetchPackageInfo` and not `fetchVulnerabilityInfo` in the following example: 
<img width="840" alt="Screenshot 2024-06-25 at 4 01 16 PM" src="https://github.com/ServiceNow/vulnerability-response/assets/167475200/7866df1f-67ce-4c5b-a732-55a056c66df7">